crates/router/src/types/storage/revenue_recovery.rs

```diff
@@ -29,9 +34,12 @@ pub struct RevenueRecoveryPaymentData {
 impl RevenueRecoveryPaymentData {
     pub async fn get_schedule_time_based_on_retry_type(
         &self,
+        state: &SessionState,
         db: &dyn StorageInterface,
```

Comment:
Why do you need both `state` and `db` to be passed?

Either accept only `state`, or accept `db` and the gRPC client as parameters.